guides/source/active_support_instrumentation.md

```diff
@@ -3,623 +3,1160 @@
+```
+
+### Subscribing to a Single Event using Multiple Subscribers
+
+You can subscribe to a single event using multiple subscribers. This is useful
+if you want to subscribe to an event from multiple sources.
+
+For example, you may want to subscribe to a `publish.posts` event from multiple
+sources to kick off a background job to send an email to the post author, and to
+also log the event.
+
 
 Rails Framework Hooks
 ---------------------
 
-Within the Ruby on Rails framework, there are a number of hooks provided for common events. These events and their payloads are detailed below.
+Within the Ruby on Rails framework, there are a number of hooks provided for
+common events.
+
+Each event below lists the event name you can subscribe to, explains how the
+event is triggered, and the corresponding example `event.payload` from the
+subscribed event.
+
+To subscribe to a specific event, use
+[`ActiveSupport::Notifications.subscribe`](https://api.rubyonrails.org/classes/ActiveSupport/Notifications.html#method-c-subscribe).
+For example:
+
+```ruby
+ActiveSupport::Notifications.subscribe "process_action.action_controller" do |event|
+  # Do something with the event
+  Rails.logger.info event.name         # "process_action.action_controller"
+  Rails.logger.info event.duration     # 10 (in milliseconds)
+  Rails.logger.info event.allocations  # 1826
+  Rails.logger.info event.payload
+  #   {
+  #   controller: "PostsController",
+  #   action: "index",
+  #   params: {"action" => "index", "controller" => "posts"},
+  #   headers: #<ActionDispatch::Http::Headers:0x...>,
+  #   format: :html,
+  #   method: "GET",
+  #   path: "/posts",
+  #   request: #<ActionDispatch::Request:0x...>,
+  #   response: #<ActionDispatch::Response:0x...>,
+  #   status: 200,
+  #   view_runtime: 46.848,
+  #   db_runtime: 0.157
+  # }
+end
+```
+
+NOTE: `event.payload` is a hash with the payload of the event. Below we describe
+the keys, and examples values that you can expect to see in the payload.
+
+Read more about subscribing to the event in [Subscribing to an
+Event](#subscribing-to-an-event).
 
 ### Action Cable
 
 #### `perform_action.action_cable`
 
-| Key              | Value                     |
-| ---------------- | ------------------------- |
-| `:channel_class` | Name of the channel class |
-| `:action`        | The action                |
-| `:data`          | A hash of data            |
+The event is emitted when a channel action is invoked.
+
+For example:
+
+```ruby
+# Client sends: { "action":"speak", "message":"Hi" }
+class ChatChannel < ApplicationCable::Channel
+  def speak(data) = Message.create!(text: data["message"])
+end
+```
+
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key      | Description               | Example               |
```

Comment:
If we rename the column header to "Example Value" (we also have "Payload Key"), the sentence can be simplified somewhat:
```suggestion
The event payload (`event.payload`) includes the following keys:

| Payload Key      | Description               | Example Value   |
```